test/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScannerUnitTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.htmlparser.jericho.Source;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscanrules.CookieHttpOnlyScanner;
+import org.zaproxy.zap.utils.I18N;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CookieHttpOnlyScannerUnitTest {
+
+	private CookieHttpOnlyScanner rule;
+	private PassiveScanThread parent;
+	private List<Alert> alertsRaised;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Constant.messages = Mockito.mock(I18N.class);
+    }
+
+	@Before
+	public void setUp() throws Exception {
+		alertsRaised = new ArrayList<>();
+		parent = new PassiveScanThread(null, null, new ExtensionAlert()) {
+			@Override
+			public void raiseAlert(int arg0, Alert arg1) {
+				alertsRaised.add(arg1);
+			}
+		};
+		rule = new CookieHttpOnlyScanner();
+		rule.setParent(parent );
+	}
+
+    @Test
+	public void noHttpOnlyAttribute () throws HttpMalformedHeaderException {
+		
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+		
+		msg.setResponseBody("<html></html>");
+		msg.setResponseHeader(
+				"HTTP/1.1 200 OK\r\n" +
+				"Server: Apache-Coyote/1.1\r\n" +
+				"Set-Cookie: test=123; Path=/;\r\n" +
+				"Content-Type: text/html;charset=ISO-8859-1\r\n" +
+				"Content-Length: " + msg.getResponseBody().length() + "\r\n");
+		String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+		Source source = new Source(response);
+		rule.scanHttpResponseReceive(msg, rule.getPluginId(), source);
```

Comment:
The seconds parameter is the ID of the history reference being scanned, probably better to use other number than the plugin ID for that.